packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -593,13 +606,32 @@ class _CupertinoBackGestureController<T> {
     // Fling in the appropriate direction.
     // AnimationController.fling is guaranteed to
     // take at least one frame.
-    if (velocity.abs() >= _kMinFlingVelocity) {
-      controller.fling(velocity: -velocity);
-    } else if (controller.value <= 0.5) {
-      controller.fling(velocity: -1.0);
+    //
+    // This curve has been determined through rigorously eyeballing native iOS
+    // animations.
+    const Curve animationCurve = Curves.fastLinearToSlowEaseIn;
+    bool animateForward;
+
+    // If the user releases the page before mid screen with sufficient velocity,
+    // or after mid screen, we should animate the page out. Otherwise, the page
+    // should be animated back in.
+    if (velocity.abs() >= _kMinFlingVelocity)
+      animateForward = velocity > 0 ? false : true;
+    else
+      animateForward = controller.value > 0.5 ? true : false;
+
+    if (animateForward) {
+      // The closer the panel is to dismissing, the shorter the animation is.
+      // We want to cap the animation time, but we want to use a linear curve
+      // to determine it.
+      final int milliseconds = min(lerpDouble(_kMaxPageForwardAnimationTime, 0, controller.value).floor(),
+                                   _kMaxPageBackAnimationTime);
+      controller.animateTo(1.01, duration: Duration(milliseconds: milliseconds), curve: animationCurve);
```

Comment:
oh I see. Why 1.01? This definitely needs some code comments

Reply:
So if you look at this: https://github.com/flutter/flutter/blob/649f49d4bfeed3f9bbcdaaf301da85fd91671f3c/packages/flutter/lib/src/animation/animation_controller.dart#L584, you can see that for fling they added or subtracted .01 from the value. So I thought it would make sense to do that for the replacement animation? But maybe that value is just so that the simulation finishes? You think it would be ok to remove the .01? I don't think the graphic difference is noticeable.

Reply:
They should have added a test :D. If we can't figure out why it was added, we should remove it.